test/parameters.jl

```diff
@@ -6,13 +6,17 @@ pdiagmat(args...) = PDiagMat(args...)
 @testset "parameters" begin
 
     @testset "postive" begin
-        @testset "$val" for val in [5.0, 1e-11, 1e-12]
+        @testset "$val" for val in [5.0, 0.001f0, 1.0e-7]
             p = positive(val)
             test_parameter_interface(p)
             @test value(p) ≈ val
+            @test typeof(value(p)) === typeof(val)
```

Comment:
Test that value remains the same. This is the main thing we want to ensure here.